### PR TITLE
Adding host metadata to the inventory payload

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Adding host metadata to the inventory payload. This will be used in the backend to populate the `inventories` product.

### Describe how to test/QA your changes

Start the Agent and wait for the inventory payload to be sent (this can be configured to a shorter time using  `DD_INVENTORIES_MIN_INTERVAL` and `DD_INVENTORIES_MAX_INTERVAL`.

Then check that the payload was sent successfully and that it contains the correct values. `datadog-agent troubleshooting metadata_inventory` command can be use to show the latest payload sent.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by olivielpeau** [comment]
This PR makes me think of something else actually: when `enable_metadata_collection` is disabled, I think we want the Agent _not to_ send inventories payloads either (because, similar to the "legacy" metadata payloads, the inventories payloads should only be sent by at most one agent per host). WDYT? If you agree, can you make sure this is ensured today?

**Comment by hush-hush** [comment]
> This PR makes me think of something else actually: when `enable_metadata_collection` is disabled, I think we want the Agent _not to_ send inventories payloads either (because, similar to the "legacy" metadata payloads, the inventories payloads should only be sent by at most one agent per host). WDYT? If you agree, can you make sure this is ensured today?

It's a good question: the `agent_metadata` and `check_metadata` might be different from agent to agent and we might want them. The `host_metadata` should only be sent once. I think we need finer control for this. WDYT ?

File: pkg/metadata/inventories/README.md [hunk 1]
```diff
@@ -74,6 +86,31 @@ The payload is a JSON dict with the following fields
...
+  - `cpu_frequency` - **number/float**: the CPU frequency.
+  - `cpu_cache_size` - **int**: the CPU cache size in bytes (only fill on Linux only, 0 for Windows and OSX).
+  - `cpu_architecture` - **string**: the hardware name, Linux only (ex "x86_64", "unknown", ...).
+  - `kernel_name` - **string**: the kernel name (ex: "windows", "Linux", ...).
+  - `kernel_release` - **string**:  the kernel release (ex: "10.0.20348", "4.15.0-1080-gcp", ...).
+  - `kernel_version` - **string**:  the kernel version (Unix only, empty string on Windows).
+  - `os` - **string**: the OS name description (ex: "GNU/Linux", "Windows Server 2022 Datacenter", ...).
+  - `os_version` - **string**: the OS version (ex: "debian bookworm/sid", ...).
+  - `python_version` - **string**: The Python version from the agent environment. `python -V` is used in `Gohai` for
+    this. Unless the Agent environment has been modified this is the Python version from the OS not from the Agent.
```
Comment (djmitche) [inline pkg/metadata/inventories/README.md:106]: This seems contradictory ("agent environment" vs. "from the OS").  I suspect this value will be confusing as often as it's useful.  Would it make sense to just omit it?

File: pkg/metadata/inventories/README.md [hunk 2]
```diff
@@ -74,6 +86,31 @@ The payload is a JSON dict with the following fields
...
+  - `memory_total_kb` - **int**: the total memory size for the host in kb.
+  - `memory_swap_total_kb` - **int**: the `swap` memory size in kb (Unix only).
```
Comment (djmitche) [inline pkg/metadata/inventories/README.md:108]:
```suggestion
  - `memory_total_kb` - **int**: the total memory size for the host in KiB.
  - `memory_swap_total_kb` - **int**: the `swap` memory size in KiB (Unix only).
```

**Review by djmitche — APPROVED** [review]

File: pkg/metadata/inventories/README.md [hunk 3]
```diff
@@ -74,6 +86,31 @@ The payload is a JSON dict with the following fields
...
+  - `ip_address` - **string**: the IP address for th host.
+  - `ipv6_address` - **string**: the IPV6 address for th host.
+  - `mac_address` - **string**: the MAC address for th host.
+  - `agent_version` - **string**: the version of the Agent.
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply pkg/metadata/inventories/README.md:112]: In the future each section might be sent in different payload (due to size limitation in the intake). We need the version of the agent in the backend to parse the payload and potentially exclude some versions.

**Review by mbotarro — APPROVED** [review]

**Pull request merged**